helix-core/src/transaction.rs

```diff
@@ -534,6 +535,46 @@ impl Transaction {
         Self::from(changeset)
     }
 
+    /// Generate a transaction from a set of potentially overallping deletions
+    /// by merging overlapping deletions together.
+    pub fn delete<I>(doc: &Rope, changes: I) -> Self
```

Comment:
very minor nit but I think this would be clearer as `deletions` rather than `changes`

Reply:
I have fixed that in the force push